### DISCUSSION TITLE
Best practices for using lint in extensions

### DISCUSSION THREAD

**Discussion by LuizHAP** [directus/directus#11036] [Feature Requests]
I created some extensions for my project using directus, and what is the best practice to configure eslint, prettier and husky for extensions in Directus?

I thought about creating a file in the root, and extending it using Shareable Configs (https://eslint.org/docs/developer-guide/shareable-configs). Is it best practice? Can I have this same thought with the prettier and husky?

The idea is to put the directus inside a monorepo with the turborepo is an option too

**Comment by nickrum** [comment]
I would recommend separating your extensions from your main Directus project. Using a monorepo for this sounds like a very good idea. Then you could either create a single config file for eslint, prettier and husky in the root of your monorepo and extend them for each individual extension as needed. Alternatively, you could create separate configs for API and App extensions in the root and use them accordingly. This would reduce duplication, as the eslint and prettier rules will likely not change much between extensions of the same type.
